test_scripts/AppServices/RPCPassThrough/001_SendLocation_Handled_Request.lua

```diff
@@ -0,0 +1,104 @@
+--  Precondition: 
+--  1) app1 and app2 are registered on SDL.
+--  2) AppServiceProvider permissions(with NAVIGATION AppService permissions to handle rpc SendLocation) are assigned for <app1ID>
+--  4) SendLocation permissions are assigned for <app2ID>
+--
+--  Steps:
+--  1) app1 sends a PublishAppService (with {serivceType=NAVIGATION, handledRPC=SendLocation} in the manifest)
+--  2) app2 sends a SendLocation request to core
+--
+--  Expected:
+--  1) Core forwards the request to app1
+--  2) app1 responds to core with { success = true, resultCode = "SUCCESS", info = "Request was handled by app services" }
+--  3) Core forwards the response from app1 to app2
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/AppServices/commonAppServices')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local functions ]]
+local function PTUfunc(tbl)
+    local pt_entry = common.getAppServiceProducerConfig(1)
+    pt_entry.app_services.NAVIGATION = { handled_rpcs = {{function_id = 39}} }
+    tbl.policy_table.app_policies[common.getConfigAppParams(1).fullAppID] = pt_entry
+end
+local function PTUfunc2(tbl)
+    local pt_entry = common.getAppServiceProducerConfig(2)
+    pt_entry.groups = { "Base-4" , "SendLocation" }
+    tbl.policy_table.app_policies[common.getConfigAppParams(2).fullAppID] = pt_entry;
+end
+
+local function RPCPassThruTest(rpc, expectedResponse, passThruResponse)
+    local providerMobileSession = common.getMobileSession(1)
+    local MobileSession = common.getMobileSession(2)
+    
+    local cid = MobileSession:SendRPC(rpc.name, rpc.params)
+        
+    providerMobileSession:ExpectRequest(rpc.name, rpc.params):Do(function(_, data)
+        providerMobileSession:SendResponse(rpc.name, data.rpcCorrelationId, passThruResponse)
+    end)
+    
+    if rpc.hmi_name then
+        EXPECT_HMICALL(rpc.hmi_name, requestParams):Times(0)
+    end
+
+    MobileSession:ExpectResponse(cid, expectedResponse.params)
+end
+
+--[[ Local variables ]]
+local manifest = {
+    serviceName = config.application1.registerAppInterfaceParams.appName,
+    serviceType = "NAVIGATION",
+    handledRPCs = {39},    
+    allowAppConsumers = true,
+    rpcSpecVersion = config.application1.registerAppInterfaceParams.syncMsgVersion,
+    navigationServiceManifest = {}
+}
+
+local successResponse = {
+    success = true,
+    resultCode = "SUCCESS",
+    info = "Request was handled by app services"
+}
+
+local sendLocationRequest = { 
+        name = "SendLocation",
+        hmi_name = "Navigation.SendLocation",
+        params = {
+            longitudeDegrees = 50,
+            latitudeDegrees = 50,
+            locationName = "TestLocation" 
+        },
+        hmi_params = params
```

Comment:
fixed the indentation. As for the `hmi_name` I am using to check that there are no calls to the hmi(since the ASP is handling the request) 
```
if rpc.hmi_name then
    EXPECT_HMICALL(rpc.hmi_name, requestParams):Times(0)
  end
```